src/utils/Encode.sol

```diff
@@ -0,0 +1,156 @@
+        A
+    }
+
+    EncProxyAdmin internal constant proxyAdmin = EncProxyAdmin.A;
+    EncUpgradeableBeacon internal constant upgradeableBeacon = EncUpgradeableBeacon.A;
+    EncGnosisSafe internal constant gnosisSafe = EncGnosisSafe.A;
+
+    /// @dev Creates a new, clean `MultisigCall[] storage` pointer, guaranteeing
+    /// any previous pointers will not be overwritten.
+    /// Since we're in a library, we have to use assembly+slot pointers, but the
+    /// high level version of this function equates to:
+    ///
+    /// uint _idx = storage.idx;
+    /// storage.idx++;
+    /// MultisigCall[] storage calls = storage.map[_idx];
+    /// return calls;
+    function newMultisigCalls() internal returns (MultisigCall[] storage) {
+        bytes32 _IDX_SLOT = IDX_SLOT;
+        bytes32 _MAP_SLOT = MAP_SLOT;
+
+        uint256 idx;
+        assembly {
+            idx := sload(_IDX_SLOT)
+            sstore(_IDX_SLOT, add(1, idx))
+        }
+
+        // fn pointer indirection fools the compiler into letting us have
+        // an uninitialized storage pointer
+        function() pure returns (mapping(uint => MultisigCall[]) storage) fn;
+        function() pure returns (uint) fn2 = func;
+        assembly {
+            fn := fn2
+        }
+        mapping(uint256 => MultisigCall[]) storage map = fn();
+        assembly {
+            map.slot := _MAP_SLOT
+        }
+
+        return map[idx];
+    }
+
+    function func() internal pure returns (uint256) {
+        return 0;
+    }
+
+    /// @dev Appends a call to a list of `MultisigCalls`, returning the original storage pointer
+    /// to facilitate call chaining syntax, e.g:
+    ///
+    /// calls
+    ///   .append(...)
+    ///   .append(...)
+    ///   .append(...);
+    function append(MultisigCall[] storage calls, address to, bytes memory data)
+        internal
+        returns (MultisigCall[] storage)
+    {
+        calls.push(MultisigCall({to: to, value: 0, data: data}));
+
+        return calls;
+    }
+
+    /// @dev Encodes a call to `ProxyAdmin.upgrade(proxy, impl)`
+    function upgrade(EncProxyAdmin, address proxy, address impl) internal pure returns (bytes memory) {
+        return abi.encodeCall(IProxyAdmin.upgrade, (ITransparentUpgradeableProxy(proxy), impl));
+    }
+
+    /// @dev Encodes a call to `UpgradeableBeacon.upgradeTo(newImpl)`
+    function upgradeTo(EncUpgradeableBeacon, address newImpl) internal pure returns (bytes memory) {
+        return abi.encodeCall(IUpgradeableBeacon.upgradeTo, (newImpl));
+    }
+
+    /// @dev Encodes a call to `MultiSend.multiSend(data)`
+    function multiSend(MultisigCall[] memory calls) internal pure returns (bytes memory) {
+        bytes memory packedCalls = new bytes(0);
+
+        for (uint256 i = 0; i < calls.length; i++) {
+            packedCalls = abi.encodePacked(
+                packedCalls,
+                abi.encodePacked(uint8(0), calls[i].to, calls[i].value, uint256(calls[i].data.length), calls[i].data)
```

Comment:
would normally link to https://github.com/safe-global/safe-smart-account/blob/main/contracts/libraries/MultiSend.sol#L20 for this gibberish xD